### PR TITLE
Feature/df 2257 kae form

### PR DESCRIPTION
**Pull request by ganqqwerty**

### CODE REVIEW

File: alv-portal-ui/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -190,6 +188,15 @@ const appRoutes: Routes = [
       appContext: AppContext.EALV
     }
   },
+  {
+    path: 'showcase/multifile-demo',
```
Comment (emina-str) [inline alv-portal-ui/src/app/app-routing.module.ts:192]: think we don't need that, right?

File: alv-portal-ui/src/app/app.module.ts [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
-import { ShowcaseComponent } from './showcase/showcase.component';
 import { WINDOW_PROVIDERS } from './core/window.service';
-import { BlockUIModule } from 'ng-block-ui';
+import { SharedModule } from './shared/shared.module';
+import { FormMultiFileUploadDemoComponent } from './showcase/form-multi-file-upload-demo/form-multi-file-upload-demo.component';
+import { ShowcaseComponent } from './showcase/showcase.component';
 
 @NgModule({
-  declarations: [AppComponent, ShowcaseComponent],
+  declarations: [AppComponent, ShowcaseComponent, FormMultiFileUploadDemoComponent],
```
Comment (emina-str) [inline alv-portal-ui/src/app/app.module.ts:17]: This one as well? FormMultiFileUploadDemoComponent? Could it be removed?

File: alv-portal-ui/src/app/online-services/kae/kae/kae-form-mapper.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  reasonForShortTimeWork: string;
+  measuresToAvoidShortTimeWork: string;
+  postponedOrders: string;
+  confirmation: boolean;
+
+}
+
+function mapAddressToDto(companyAddress: PostAddressFormValue): AddressDto {
+  return {
+    canton: '', //todo
```
Comment (emina-str) [inline alv-portal-ui/src/app/online-services/kae/kae/kae-form-mapper.ts:52]: I've removed it, you can as well. Thanks for checking with me

File: alv-portal-ui/src/app/online-services/kae/kae/kae-form-mapper.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  };
+}
+
+function mapContractNumbersFormValueToDto(validFormValue: ContractNumbersFormValue): ContractNumbersDto {
+  return {
+    unlimitedEmploymentContracts: Number(validFormValue.unlimitedEmploymentContracts),
+    terminatedEmploymentContracts: Number(validFormValue.terminatedEmploymentContracts),
+    limitedEmploymentContracts: Number(validFormValue.limitedEmploymentContracts),
+    onCallContracts: Number(validFormValue.onCallContracts),
+    internshipContracts: Number(validFormValue.internshipContracts)
```
Comment (emina-str) [inline alv-portal-ui/src/app/online-services/kae/kae/kae-form-mapper.ts:70]: we do expect also "total" from frontend.

File: alv-portal-ui/src/app/online-services/kae/kae/kae-form-mapper.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    companyEstablishmentDate: toISOLocalDate(formValue.companyEstablishmentDate),
+    reasonForLessOrders: formValue.reasonForLessOrders,
+    monthlyTurnover: formValue.monthlyTurnover,
+    orderBacklog: formValue.orderBacklog,
+    businessProjection: formValue.businessProjection,
+    reasonForShortTimeWork: formValue.reasonForShortTimeWork,
+    measuresToAvoidShortTimeWork: formValue.measuresToAvoidShortTimeWork,
+    postponedOrders: formValue.postponedOrders
+  };
+  if (formValue.concernType === ConcernTypeFormValue.COMPANY) {
```
Comment (emina-str) [inline alv-portal-ui/src/app/online-services/kae/kae/kae-form-mapper.ts:100]: the new specs state that for either case of concernType the `companyContractsCurrent` and `companyContractsLastYear` must be filled out.
Only for `DEPARTMENT `is then the `departmentContractsCurrent` and `departmentContractsLastYear` needed.

File: alv-portal-ui/src/app/online-services/kae/kae/kae-form-mapper.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    ahvAccountNumber: formValue.ahvAccountNumber,
+    companyActivity: formValue.companyActivity,
+    companyEstablishmentDate: toISOLocalDate(formValue.companyEstablishmentDate),
+    reasonForLessOrders: formValue.reasonForLessOrders,
+    monthlyTurnover: formValue.monthlyTurnover,
+    orderBacklog: formValue.orderBacklog,
+    businessProjection: formValue.businessProjection,
+    reasonForShortTimeWork: formValue.reasonForShortTimeWork,
+    measuresToAvoidShortTimeWork: formValue.measuresToAvoidShortTimeWork,
+    postponedOrders: formValue.postponedOrders
```
Comment (emina-str) [inline alv-portal-ui/src/app/online-services/kae/kae/kae-form-mapper.ts:98]: missing `reasonWhyWorkLossIsTemporary` (question 12)

File: alv-portal-ui/src/app/online-services/kae/kae/kae.component.ts [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+      concernType: ['', Validators.required],
+      departmentName: ['', [Validators.required, Validators.maxLength(50)]],
+      shortTimeWorkCount: ['', Validators.required],
+      workLossPercentage: ['', [Validators.required, Validators.min(0), Validators.max(100)]],
+      holidaysIncluded: ['', Validators.required],
+      unemploymentInsuranceFund: ['', [Validators.required, Validators.maxLength(50)]],
+      ahvCompensationFund: ['', Validators.required, Validators.maxLength(50)],
+      ahvAccountNumber: ['', Validators.required, Validators.maxLength(50)],
+      companyActivity: ['', Validators.required, Validators.maxLength(500)], //fixme here will be smart multiline validator
+      companyEstablishmentDate: ['', Validators.required, Validators.maxLength(500)], //fixme here will be smart multiline validator
```
Comment (emina-str) [inline alv-portal-ui/src/app/online-services/kae/kae/kae.component.ts:102]: companyEstablishmentDate don't need maxLength validator

File: alv-portal-ui/src/app/online-services/kae/kae/kae.component.ts [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+      reasonForLessOrders: ['', Validators.required, Validators.maxLength(500)], //fixme here will be smart multiline validator
+      monthlyTurnover: ['', Validators.required, Validators.maxLength(500)], //fixme here will be smart multiline validator
+      orderBacklog: ['', Validators.required, Validators.maxLength(500)], //fixme here will be smart multiline validator
+      businessProjection: ['', Validators.required, Validators.maxLength(500)], //fixme here will be smart multiline validator
+      reasonForShortTimeWork: ['', Validators.required, Validators.maxLength(500)], //fixme here will be smart multiline validator
+      measuresToAvoidShortTimeWork: ['', Validators.required, Validators.maxLength(500)], //fixme here will be smart multiline validator
+      postponedOrders: ['', Validators.required, Validators.maxLength(500)], //fixme here will be smart multiline validator
```
Comment (emina-str) [inline alv-portal-ui/src/app/online-services/kae/kae/kae.component.ts:109]: 12 question : reasonWhyWorkLossIsTemporary : seems to be missing.

File: alv-portal-ui/src/app/online-services/kae/post-address-form/post-address-form.component.ts [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+@Component({
+  selector: 'alv-post-address-form',
+  templateUrl: './post-address-form.component.html',
+  styleUrls: ['./post-address-form.component.scss']
+})
+export class PostAddressFormComponent implements OnInit, OnDestroy {
+
+  readonly PO_BOX_MAX_LENGTH = 6;
+  readonly HOUSE_NUMBER_MAX_LENGTH = 10;
+  readonly STREET_MAX_LENGTH = 60;
```
Comment (emina-str) [inline alv-portal-ui/src/app/online-services/kae/post-address-form/post-address-form.component.ts:21]: max value for street is 50, and also for name 50.

File: alv-portal-ui/src/app/shared/backend-services/kae/kae.types.ts [hunk 10]
```diff
@@ -0,0 +1,86 @@
+export interface AddressDto {
+  name: string;
+  street: string;
+  houseNumber: string;
+  poBox: string;
+  postalCode: string;
+  city: string;
+  canton: string;
```
Comment (emina-str) [inline alv-portal-ui/src/app/shared/backend-services/kae/kae.types.ts:8]: you can remove canton, thanks.

File: alv-portal-ui/src/app/shared/backend-services/kae/kae.types.ts [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+export interface KaeDto {
+  companyAddress: AddressDto;
+  branch: string;
+  advisor: string;
+  companyEmail: string;
+  companyPhone: string;
+  concernType: 'COMPANY' | 'DEPARTMENT';
+  departmentName?: string;
+  companyContractsCurrent?: ContractNumbersDto;
```
Comment (emina-str) [inline alv-portal-ui/src/app/shared/backend-services/kae/kae.types.ts:28]: companyContractsCurrent and companyContractsLastYear are mandatory

Comment (ganqqwerty) [reply alv-portal-ui/src/app/shared/backend-services/kae/kae.types.ts:28]: they are mandatory if you picked concernType: 'COMPANY'

Comment (emina-str) [reply alv-portal-ui/src/app/shared/backend-services/kae/kae.types.ts:27]: no, they are always mandatory.

> Re: eALV-KAE-04 - UC: Formular Frage 1-4

@julien studzinski @Iurii Katkov 
I just received the information that I had wrong information. If you click “Betriebsabteilung” you have to fill out 2a - 2d. If it is “Gesamtbetrieb” you only have answer 2a and 2b.

Comment (ganqqwerty) [reply alv-portal-ui/src/app/shared/backend-services/kae/kae.types.ts:27]: I can kinda express it in types, but not now

Comment (emina-str) [reply alv-portal-ui/src/app/shared/backend-services/kae/kae.types.ts:27]: what do you mean, you can't implement it today? we have validation for mandatory fields on backend ..

**Comment by emina-str** [comment]
@ganqqwerty   I created a release branch for this.

**Pull request closed without merging**